src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -182,29 +196,82 @@ def _list_services(self, cmd):
                     s.version,
                     s.rados_config_location))
 
-            return HandleCommandResult(stdout="\n".join(lines))
+            return HandleCommandResult(odata="\n".join(lines))
 
-    def _service_add(self, cmd):
+    def _create_osd(self, inbuf, cmd):
+        """Create one or more OSDs
+
+        :cmd : Arguments for the create osd
+        """
+        #Obtain/validate parameters for the operation
+        cmdline_error = ""
+        if "drive_group" in cmd.keys():
+            params = cmd["drive_group"]
+        elif inbuf:
+            params = inbuf
+        else:
+            cmdline_error = "Please, use 'drive_group' parameter \
+                             or specify -i <input file>"
+
+        if cmdline_error:
+            return HandleCommandResult(-errno.EINVAL, rs=cmdline_error)
+
+        drive_group = orchestrator.DriveGroupSpec(params)
```

Comment:
```suggestion
        drive_group = orchestrator.DriveGroupSpec.from_json(params)
```

Reply:
It will be as you suggest once rebased with https://github.com/ceph/ceph/pull/25912 included.